repose-aggregator/core/core-lib/src/main/java/org/openrepose/core/services/deploy/ArtifactDirectoryWatcher.java

```diff
@@ -71,17 +71,21 @@ public synchronized void run() {
     }
 
     // TODO: decompose
-    private void checkArtifacts(final File artifactDirectoryReference) {
-        final Set<String> removedArtifacts = new HashSet<String>(artifactModificationTimes.keySet());
 
-        if (artifactDirectoryReference == null) {
+    /**
+     * @return true if artifacts were modified, false otherwise
+     */
+    public synchronized boolean checkArtifacts() {
```

Comment:
mildly concerned about having this be synchronized, especially since it's used in a while loop.